extensions/sql-bindings/src/test/service/azureFunctionsService.test.ts

```diff
@@ -193,9 +190,149 @@ describe('AzureFunctionsService', () => {
+		});
+
+		it('Should prompt select table when user cancels out of manually entering table', async function (): Promise<void> {
+			// This test will have an azure function project already in the project and the azure functions extension installed (stubbed)
+			sinon.stub(azureFunctionUtils, 'getAzureFunctionsExtensionApi').resolves(testUtils.azureFunctionsExtensionApi.object); // set azure functions extension api
+			sinon.stub(azureFunctionUtils, 'getAzureFunctionProject').resolves(projectFilePath); //set azure function project to have one project
+			sinon.stub(utils, 'getVscodeMssqlApi').resolves(testUtils.vscodeMssqlIExtension.object);
+
+			let connectionInfo: IConnectionInfo = createTestCredentials();// Mocks promptForConnection
+
+			let connectionDetails = { options: connectionInfo };
+			testUtils.vscodeMssqlIExtension.setup(x => x.getConnectionString(connectionDetails, true, false)).returns(() => Promise.resolve('testConnectionString'));
+			// select input or output binding
+			let quickpickStub = sinon.stub(vscode.window, 'showQuickPick').resolves(<any>{ label: constants.input, type: BindingType.input });
+
+			// no table used for connection info so prompt user to get connection info
+			// promptForConnection is set first time for user
+			let promptForConnectionStub = sinon.stub(testUtils.vscodeMssqlIExtension.object, 'promptForConnection').withArgs(true).onFirstCall().resolves(connectionInfo);
+			// setup listDatabases request with connectionURI
+			testUtils.vscodeMssqlIExtension.setup(x => x.connect(connectionInfo)).returns(() => Promise.resolve('testConnectionURI'));
+			testUtils.vscodeMssqlIExtension.setup(x => x.listDatabases('testConnectionURI')).returns(() => Promise.resolve(['testDb']));
+			// select the testDB from list of databases based on connection info
+			quickpickStub.onSecondCall().resolves(('testDb') as any);
+			// get tables from selected database
+			const params = { ownerUri: 'testConnectionURI', queryString: azureFunctionUtils.tablesQuery('testDb') };
+			testUtils.vscodeMssqlIExtension.setup(x => x.sendRequest(azureFunctionsContracts.SimpleExecuteRequest.type, params))
+				.returns(() => Promise.resolve({ rowCount: 1, columnInfo: [], rows: [[{ displayValue: '[schema].[testTable]' }]] }));
+			// select the to manually enter table name
+			let manuallyEnterObjectName = constants.manuallyEnterObjectName(constants.enterObjectName);
+			quickpickStub.onThirdCall().resolves(manuallyEnterObjectName as any);
+			// cancel out of manually enter inputBox
+			sinon.stub(vscode.window, 'showInputBox').resolves(undefined);
+			// resolve promises to undefined to exit out of createFunction
+			quickpickStub.onCall(4).resolves(undefined);
+			promptForConnectionStub.onSecondCall().resolves(undefined);
+
+			should(connectionInfo.database).equal('my_db', 'ConnectionInfo database should not be changed');
```

Comment:
Is this check needed here? createTestCredentials will always set the database to 'my_db'.

Reply:
I think its a good check to have prior just to ensure that we see that connectionInfo is currently set to my_db and then after certain tests, it should change to testDb.